### PR TITLE
feat: Handle missing fields in function and procedure

### PR DESCRIPTION
**Pull request by sfc-gh-asawicki**
Handle secrets, external access integrations, comments, packages, and snowpark package.

### CODE REVIEW

File: pkg/resources/procedure_java.go [hunk 1]
```diff
@@ -62,8 +62,12 @@ func CreateContextProcedureJava(ctx context.Context, d *schema.ResourceData, met
...
 	handler := d.Get("handler").(string)
 	runtimeVersion := d.Get("runtime_version").(string)
-	// TODO [this PR]: handle real packages
-	packages := []sdk.ProcedurePackageRequest{*sdk.NewProcedurePackageRequest("com.snowflake:snowpark:1.14.0")}
+
+	packages, err := parseProceduresPackagesCommon(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	packages = append(packages, *sdk.NewProcedurePackageRequest(fmt.Sprintf(`%s%s`, sdk.JavaSnowparkPackageString, d.Get("snowpark_package").(string))))
```
Comment (sfc-gh-jcieslak) [inline pkg/resources/procedure_java.go:70]: from this I'm guessing snowpark doesn't have to be at first position, but has to be included?

File: pkg/sdk/testint/functions_integration_test.go [hunk 2]
```diff
@@ -34,6 +34,7 @@ import (
 // TODO [SNOW-1348103]: test secure
 // TODO [SNOW-1348103]: python aggregate func (100357 (P0000): Could not find accumulate method in function CVVEMHIT_06547800_08D6_DBCA_1AC7_5E422AFF8B39 with handler dump)
 // TODO [SNOW-1348103]: add test with multiple imports
+// TODO [this PR]: test with multiple external access integrations and secrets
```
Comment (sfc-gh-jmichalak) [inline pkg/sdk/testint/functions_integration_test.go:37]: Next PR or is it already done?

File: pkg/sdk/procedures_ext.go [hunk 3]
```diff
@@ -108,6 +118,75 @@ func procedureDetailsFromRows(rows []ProcedureDetail) (*ProcedureDetails, error)
...
+		v.NormalizedSecrets = map[string]SchemaObjectIdentifier{}
+	}
+
+	if v.Packages != nil {
+		if p, err := parseFunctionOrProcedurePackages(*v.Packages); err != nil {
+			errs = append(errs, err)
+		} else {
+			// TODO [SNOW-1850370]: merge these and unit test
+			switch strings.ToUpper(v.Language) {
+			case "JAVA", "SCALA":
```
Comment (sfc-gh-jmichalak) [inline pkg/sdk/procedures_ext.go:147]: Nit: use these as enums values. Also, add a comm explaining what we're doing here, it's not clear.

File: pkg/sdk/procedures_ext.go [hunk 4]
```diff
@@ -108,6 +118,75 @@ func procedureDetailsFromRows(rows []ProcedureDetail) (*ProcedureDetails, error)
...
+				v.NormalizedPackages = filtered
+				if !found {
+					errs = append(errs, fmt.Errorf("could not parse package from Snowflake, expected at least snowpark package, got %v", filtered))
+				}
+			case "PYTHON":
+				filtered := make([]string, 0)
+				var found bool
+				for _, o := range p {
+					o := strings.TrimSpace(o)
+					if strings.HasPrefix(o, PythonSnowparkPackageString) {
```
Comment (sfc-gh-jmichalak) [inline pkg/sdk/procedures_ext.go:168]: Nit: move the switch here, so there's less code duplication.

**Review by sfc-gh-jcieslak — APPROVED** [review]

File: pkg/resources/procedure_java.go [hunk 5]
```diff
@@ -173,11 +175,32 @@ func UpdateContextProcedureJava(ctx context.Context, d *schema.ResourceData, met
 	setRequest := sdk.NewProcedureSetRequest()
 	unsetRequest := sdk.NewProcedureUnsetRequest()
 
-	// TODO [SNOW-1348103]: handle all updates
-	// secure
-	// external access integration
-	// secrets
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/resources/procedure_java.go:179]: is secure for procedures cannot be used in alter sadeg https://docs.snowflake.com/en/sql-reference/sql/alter-procedure#java-handler

File: pkg/sdk/testint/functions_integration_test.go (hunk 2 above)
Comment (sfc-gh-asawicki) [reply pkg/sdk/testint/functions_integration_test.go:37]: it's tested on the resource level, so we are fine but I would also like to add it here (so yeah, one of the next PRs - will add ticket too)

File: pkg/sdk/procedures_ext.go (hunk 3 above)
Comment (sfc-gh-asawicki) [reply pkg/sdk/procedures_ext.go:147]: +1

File: pkg/sdk/procedures_ext.go (hunk 4 above)
Comment (sfc-gh-asawicki) [reply pkg/sdk/procedures_ext.go:168]: oke

File: pkg/resources/procedure_java.go (hunk 1 above)
Comment (sfc-gh-asawicki) [reply pkg/resources/procedure_java.go:70]: it seems so, I will check in other resources too

**Pull request merged**